### PR TITLE
Store full IndexedAttestation and SignedBeaconBlockHeader for slasher 

### PR DESCRIPTION
**Pull request by 0xKiwi**
This PR changes the new slasher to store attestations and beacon block headers as full objects, as it needs the full data of historical attestations/block headers to submit a proper slashing proof to the beacon node.

Theres a nice optimization by encoding the data with the signing root. We can easily verify if a slashing is valid by comparing the signing root without having to unmarshal the full object!

### CODE REVIEW

File: beacon-chain/slasher/detect_blocks.go [hunk 1]
```diff
@@ -3,29 +3,32 @@ package slasher
...
+	proposedBlocks []*slashpb.SignedBlkHeaderWrapper,
 ) error {
 	ctx, span := trace.StartSpan(ctx, "Slasher.detectSlashableBlocks")
 	defer span.End()
 	// We check if there are any slashable double proposals in the input list
 	// of proposals with respect to each other.
 	existingProposals := make(map[string][32]byte)
 	for i, proposal := range proposedBlocks {
-		key := uintToString(uint64(proposal.Slot)) + ":" + uintToString(uint64(proposal.ProposerIndex))
+		key := uintToString(uint64(proposal.SignedBlockHeader.Header.Slot)) + ":" + uintToString(uint64(proposal.SignedBlockHeader.Header.ProposerIndex))
```
Comment (rauljordan) [inline beacon-chain/slasher/detect_blocks.go:25]: This should be a helper func now

File: beacon-chain/slasher/receive_test.go [hunk 2]
```diff
@@ -324,45 +218,17 @@ func TestSlasher_receiveAttestations_OK(t *testing.T) {
...
-			AttestingIndices: att2.AttestingIndices,
-			Source:           att2.Data.Source.Epoch,
-			Target:           att2.Data.Target.Epoch,
-		},
+	wanted := []*slashertypes.IndexedAttestationWrapper{
+		createAttestationWrapper(att1.Data.Source.Epoch, att1.Data.Source.Epoch, att1.AttestingIndices, nil),
+		createAttestationWrapper(att2.Data.Source.Epoch, att2.Data.Source.Epoch, att2.AttestingIndices, nil),
 	}
-	require.DeepEqual(t, wanted, s.attestationQueue)
+	require.DeepSSZEqual(t, wanted, s.attestationQueue)
```
Comment (0xKiwi) [inline beacon-chain/slasher/receive_test.go:231]: Made this change as this is a pointer now

File: beacon-chain/slasher/chunks_test.go [hunk 3]
```diff
@@ -90,7 +90,7 @@ func TestMinSpanChunksSlice_CheckSlashable(t *testing.T) {
 	validatorIdx := types.ValidatorIndex(1)
 	source := types.Epoch(1)
 	target := types.Epoch(2)
-	att := createAttestation(source, target)
+	att := createAttestationWrapper(source, target, nil, nil)
```
Comment (rauljordan) [inline beacon-chain/slasher/chunks_test.go:93]: small comment to explain what the nils are could be helpful, something like (source, target, nil /* indices */, nil /* signing root */)

File: beacon-chain/slasher/detect_attestations_test.go [hunk 4]
```diff
@@ -111,7 +114,7 @@ func Test_applyAttestationForValidator_MinSpanChunk(t *testing.T) {
 	// We apply attestation with (source 1, target 2) for our validator.
 	source := types.Epoch(1)
 	target := types.Epoch(2)
-	att := createAttestation(source, target)
+	att := createAttestationWrapper(source, target, nil, nil)
```
Comment (rauljordan) [inline beacon-chain/slasher/detect_attestations_test.go:117]: +1 here

**Review by rauljordan — COMMENTED** [review]
Everything here looks great, there was just a warning from deepsource on a bug risk to look into, then this is pretty much ready to go

**Review by rauljordan — APPROVED** [review]

**Pull request merged**